### DISCUSSION TITLE
python 3.12

### DISCUSSION THREAD

**Discussion by SinicinDenis** [pokepetter/ursina#637] [General]
hi guys, I have a question about python 3.12 + ursina is not working for some reason. and another question, will Ursina project develop? and what to expect in the future? thanks

**Comment by xandaaah** [comment]
Python 3.12 is being worked on for Panda3D currently, so just downgrade to the latest Python 3.11 release.
(If you don't know what Panda3D is, ursina has it as a dependency)

**Reply by SinicinDenis** [reply]
Do you know how to create several lamps? and how to rotate along the axes? I'm talking about a light like SpotLights

**Reply by xandaaah** [reply]
> Do you know how to create several lamps? and how to rotate along the axes? I'm talking about a light like SpotLights

I don't know, I am pretty new to Ursina.

**Comment by Raphi-2Code** [comment]
install the dev version:
pip install -U --pre --extra-index-url https://archive.panda3d.org/ panda3d
